### PR TITLE
Add build step for webpack

### PR DESCRIPTION
**Pull request by hannahgooding**

### CODE REVIEW

File: webpack.config.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
+  output: {
+    path: path.resolve(__dirname, 'dist'),
+    filename: 'updated-object-diff.js',
```
Comment (matthewlawson) [inline webpack.config.js]:
```suggestion
    filename: 'updated-object-diff.js',
    globalObject: 'this',
```

Comment (matthewlawson) [reply webpack.config.js]: I'm not sure why this isn't in the webpack library author guide but I had to add this to make it work as a node module.

File: webpack.config.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+const path = require('path');
+
+module.exports = {
+  entry: './src/index.js',
```
Comment (matthewlawson) [inline webpack.config.js]: This needs the `index.js` moving to a `src` dir

**Review by matthewlawson — COMMENTED** [review]
Looks really good. I had to do a few things to get it working in node.
* The suggestions in the review
* Add `"main": "dist/updated-object-diff.js",` to package.json

Also for good housekeeping:
* add a  script to `package.json` `"prepare": "webpack"`. This will run each time you publish. 
* Add `dist` to `.gitignore`

File: webpack.config.js (hunk 2 above)
Comment (hannahgooding) [reply webpack.config.js]: I'll update the path instead 👍

**Review by matthewlawson — APPROVED** [review]
Looks good, I tested in a small "Create react app" app & directly using node, the module compile works well for both environments.

**Pull request merged**